test/manipulation.test.js

```diff
@@ -373,6 +637,22 @@ describe('manipulation', function () {
       });
     });
 
+    it('should preserve properties with dynamic setters on update (promise variant)', function(done) {
+      StubUser.create({ password: 'foo' })
+      .then(function(created) {
+        var data = { id: created.id, password: 'bar' };
+        return StubUser.updateOrCreate(data)
+        .then(function(updated) {
+          updated.password.should.equal('bar-2');
+          return StubUser.findById(created.id)
+          .then(function(found) {
+            found.password.should.equal('bar-2');
+            done();
+          });
+        });
+      }).catch(done);
+    });
```

Comment:
"preserve properties with dynamic setter" is not related to callback vs. promise, can we omit this test?